src/server/location.rs

```diff
@@ -58,6 +58,15 @@ impl Provider for TestHeaderProvider {
                 }
             }
 
+            if let Some(dma) = parts.next() {
```

Comment:
Ah, I see what you're doing. I wonder if we might want to make this more a JSON blob so there's no threat of screwing up order or something. That could definitely be a different ticket, though.